### PR TITLE
FIX: Aligment zero length sequence bug

### PR DESCRIPTION
**Pull request by pablogs9**

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Linux/12/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Windows/11/)

File: test/serialization/SequenceSerialization.hpp [hunk 1]
```diff
@@ -182,6 +182,31 @@ class SequenceSerialization : public ArraySerialization
...
+        char str_input[23] = "This is a message test";
+        char str_output[23] = {0};
+
+        EXPECT_TRUE(ucdr_serialize_string(&writer, str_input));
+        EXPECT_TRUE(ucdr_serialize_sequence_double(&writer, input, 0));
+        EXPECT_TRUE(ucdr_serialize_uint32_t(&writer, 42));
+        
+        EXPECT_TRUE(ucdr_deserialize_string(&reader, str_output, 23));
+        EXPECT_TRUE(ucdr_deserialize_sequence_double(&reader, output, 50, &output_size));
+        EXPECT_EQ(output_size, (uint32_t) 0);
```
Comment (jamoralp) [inline test/serialization/SequenceSerialization.hpp:200]:
```suggestion
        EXPECT_EQ(output_size, static_cast<uint32_t>(0));
```

File: test/serialization/SequenceSerialization.hpp [hunk 2]
```diff
@@ -182,6 +182,31 @@ class SequenceSerialization : public ArraySerialization
...
+
+        uint32_t readed;
+        EXPECT_TRUE(ucdr_deserialize_uint32_t(&reader, &readed));
+        EXPECT_EQ(readed, (uint32_t) 42);
```
Comment (jamoralp) [inline test/serialization/SequenceSerialization.hpp:204]:
```suggestion
        EXPECT_EQ(readed, static_cast<uint32_t>(42));
```

**Review by jamoralp — CHANGES_REQUESTED** [review]

File: test/serialization/SequenceSerialization.hpp (hunk 1 above)
Comment (pablogs9) [reply test/serialization/SequenceSerialization.hpp]: Please check the new approach

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Linux/15/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Windows/14/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Linux/16/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Windows/15/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Linux/17/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Windows/16/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Linux/18/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-CDR%20Manual%20Windows/17/)

**Review by jamoralp — APPROVED** [review]

**Comment by jamoralp** [comment]
Looks good. We should take into consideration making a more complete test set (I.E. testing all basic types, creating custom complex types and test against uCDR-u-CDR, FastCDR-uCDR, uCDR-FastCDR).

**Pull request merged**